server/sonar-server/src/test/java/org/sonar/server/permission/ws/AddUserActionTest.java

```diff
@@ -193,40 +165,39 @@ public void fail_when_get_request() throws Exception {
   public void fail_when_user_login_is_missing() throws Exception {
     expectedException.expect(IllegalArgumentException.class);
 
-    ws.newPostRequest(CONTROLLER, ACTION)
+    loginAsAdmin();
+    wsTester.newPostRequest(CONTROLLER, ACTION)
       .setParam(PARAM_PERMISSION, SYSTEM_ADMIN)
       .execute();
   }
 
   @Test
   public void fail_when_permission_is_missing() throws Exception {
-    expectedException.expect(IllegalArgumentException.class);
+    expectedException.expect(NotFoundException.class);
 
-    ws.newPostRequest(CONTROLLER, ACTION)
+    loginAsAdmin();
+    wsTester.newPostRequest(CONTROLLER, ACTION)
       .setParam(PARAM_USER_LOGIN, "jrr.tolkien")
       .execute();
   }
 
   @Test
   public void fail_when_project_uuid_and_project_key_are_provided() throws Exception {
+    componentTester.insertComponent(newProjectDto(A_PROJECT_UUID).setKey(A_PROJECT_KEY));
+
     expectedException.expect(BadRequestException.class);
     expectedException.expectMessage("Project id or project key can be provided, not both.");
-    insertComponent(newProjectDto("project-uuid").setKey("project-key"));
-    commit();
 
-    ws.newPostRequest(CONTROLLER, ACTION)
+    loginAsAdmin();
+    wsTester.newPostRequest(CONTROLLER, ACTION)
       .setParam(PARAM_PERMISSION, SYSTEM_ADMIN)
-      .setParam(PARAM_USER_LOGIN, "ray.bradbury")
+      .setParam(PARAM_USER_LOGIN, user.getLogin())
       .setParam(PARAM_PROJECT_ID, "project-uuid")
       .setParam(PARAM_PROJECT_KEY, "project-key")
       .execute();
   }
 
-  private void insertComponent(ComponentDto component) {
-    dbClient.componentDao().insert(dbSession, component);
-  }
-
-  private void commit() {
-    dbSession.commit();
+  private void loginAsAdmin() {
+    userSession.login("admin").setGlobalPermissions(SYSTEM_ADMIN);
```

Comment:
FYI, when specifying the login is useless `userSession.login()` will provide a default one